gobblin-runtime/src/main/java/org/apache/gobblin/runtime/local/LocalTaskStateTracker.java

```diff
@@ -90,7 +91,8 @@ public void onTaskRunCompletion(Task task) {
       // Check the task state and handle task retry if task failed and
       // it has not reached the maximum number of retries
       WorkUnitState.WorkingState state = task.getTaskState().getWorkingState();
-      if (state == WorkUnitState.WorkingState.FAILED && task.getRetryCount() < this.maxTaskRetries) {
+      String dataQualityResult = task.getTaskState().getProp(TaskLevelPolicyChecker.TASK_LEVEL_POLICY_RESULT_KEY);
+      if (TaskLevelPolicyChecker.DataQualityStatus.FAILED.name().equals(dataQualityResult) || state == WorkUnitState.WorkingState.FAILED && task.getRetryCount() < this.maxTaskRetries) {
```

Comment:
I think we need to retry in case of either data quality failure or task failure till the retry count.. however since `&&` has higher precedence than `||`, the condition this would be interpreted as 
`if ((DataQualityStatus == FAILED) || (state == FAILED && retryCount < max))` and it would be stuck in infinite retry loop if DataQualityStatus failed. Please add parenthesis to run the intended logic